antenna/throttler.py

```diff
@@ -223,29 +223,61 @@ def match_firefox_pre_57(data):
     )
 
 
-accept_all = [
+#: This accepts crash reports for all products
+ALL_PRODUCTS = []
+
+
+#: List of supported products; these have to match the ProductName of the
+#: incoming crash report
+MOZILLA_PRODUCTS = [
+    'Firefox',
+    'Fennec',
+    'Thunderbird',
+    'SeaMonkey'
+]
+
+
+#: Rule set to accept all incoming crashes
+ACCEPT_ALL = [
     # Accept everything
     Rule('accept_everything', '*', always_match, 100)
 ]
 
 
-mozilla_rules = [
-    # Drop browser side of all multi-submission hang crashes
+#: Ruleset for Mozilla's crash collector
+MOZILLA_RULES = [
+    # Reject browser side of all multi-submission hang crashes
     Rule(
         rule_name='has_hangid_and_browser',
         key='*',
-        condition=lambda d: 'HangID' in d and d.get('ProcessType', 'browser') == 'browser',
+        condition=(
+            lambda throttler, d: 'HangID' in d and d.get('ProcessType', 'browser') == 'browser'
+        ),
         percentage=None
     ),
 
-    # Drop infobar=true crashes for certain versions
+    # Reject infobar=true crashes for certain versions of Firefox desktop
     Rule(
         rule_name='infobar_is_true',
         key='*',
         condition=match_infobar_true,
         percentage=None
     ),
 
+    # Reject all unsupported products; this does nothing if the list of
+    # supported products is empty
+    Rule(
+        rule_name='unsupported_product',
+        key='*',
```

Comment:
Have to use `'*'` here because otherwise the match code will skip this rule if there's no `ProductName`. We want it to run this rule and if there's no `ProductName`, reject the crash report.